app/src/main/java/org/xbmc/kore/host/HostConnectionObserver.java

```diff
@@ -592,9 +593,29 @@ public void onSuccess(ArrayList<GetPlaylist.GetPlaylistResult> result) {
                 if (result.isEmpty())
                     return;
 
-                for (PlaylistEventsObserver observer : playlistEventsObservers) {
-                    observer.playlistsAvailable(result);
+                boolean isPlaylistChanged = false;
+                for (GetPlaylist.GetPlaylistResult getPlaylistResult : result) {
+                    if (! isPlaylistChanged &&
+                        ! prevGetPlaylistResults.contains(getPlaylistResult)) {
+                        isPlaylistChanged = true;
+                    }
+
+                    prevGetPlaylistResults.remove(getPlaylistResult);
+                }
+
+                if (isPlaylistChanged) {
```

Comment:
Can't we just call `playlistsAvailable ` on all playlists we got? Is this just an optimization?
Not needing this would obliviate the need for the `equals` method. 
Seems to me that we just need the set difference between the playlists ids on preResults and results, so that we can call `playlistOnClear` on them, and if so this should be more explicit.

Reply:
Yes, it's an optimisation. For HTTP connections there is a chance we need to check a lot of items each time the checker is called. But I've been thinking about this a bit more and as it only benefits the case where the user changes the playlist, it benefits only a very small portion of the amount of times the `checkPlaylist` method is called. So this logic can be removed clearing up the code considerably.